analyze_logs.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+import fileinput # For accessing stdin
+
+import slack_sdk
+
+import slack_reporter_config as config
+
+
+def report_no_output():
+    client = slack_sdk.WebClient(token=config.auth_token)
+    text = ("The canary tests on the {} board had no recent output!"
+            .format(config.board_name))
+    result = client.chat_postMessage(channel=config.channel_id, text=text)
+    # If we get a result, things worked. Failure raises exceptions instead.
+    # Currently, we ignore any errors. How do we report that we're unable to
+    # report stuff!?
+
+
+def report_errors(output):
+    client = slack_sdk.WebClient(token=config.auth_token)
+    text = ("The canary tests on the {} board have failed. Recent output is:"
+        .format(config.board_name))
+    file_contents = "\n".join(output)
+
+    result = client.files_upload_v2(
+        channel=config.channel_id, content=file_contents, snippet_type="text",
+        filename="canary_tests.log", initial_comment=text)
+    # Again, if things fail we'll raise an exception here, but there is no
```

Comment:
in this scenario, could we raise an Exception whose text contains the output contents above? Or if it's too big for that maybe we could print it into another file?

Reply:
Yeah, we're writing everything into `/tmp/canary_analysis.log` (per the crontab template). but there's no obvious way to tell a human to go look at that file.

Reply:
...oh, I misunderstood! The contents of `output` are already in `/tmp/canary_tests.log`, and this script is just to send the failures over to Slack. They're still available in a file on the board itself.

Reply:
yeah that's what I meant. If the file is still available on the board, that's fine